File: src/main/java/com/history/tree/model/Marriage.java
package com.history.tree.model;


import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

@Getter
@Setter
@EqualsAndHashCode
@Table("marriage")
public class Marriage {

    @Id
    @Column("id")
    private Long id;

    @Column("date_start")
    private String dateStart;

    @Column("date_end")
    private String dateEnd;

}

File: src/main/kotlin/com/history/tree/model/Marriage.kt
package com.history.tree.model

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDate
import java.util.UUID

@Table("marriage")
data class Marriage(
    @Id
    @Column("id")
    val id: UUID,

    @Column("date_start")
    val dateStart: LocalDate?,

    @Column("date_end")
    val dateEnd: LocalDate?,

    @Column("tree_view_id")
    var treeViewId: UUID,
)